pkg/autoscaler/k8sclient/k8sclient.go

```diff
@@ -81,8 +87,28 @@ func NewK8sClient(namespace, target string, nodelabels string) (K8sClient, error
 		return nil, err
 	}
 
-	// Start propagating contents of the nodeStore.
+	restClient := clientset.RESTClient()
+	discoveryClient := discovery.NewDiscoveryClient(restClient)
 
+	// Informers don't seem to do a good job logging error messages when it
+	// can't reach the server, making debugging hard. This makes it easier to
+	// figure out if apiserver is configured incorrectly.
+	glog.Infof("Testing communication with server")
+	v, err := discoveryClient.ServerVersion()
+	if err != nil {
+		glog.Errorf("error while trying to communicate with apiserver: %s", err.Error())
+		return nil, err
+	}
+	glog.Infof("Running with Kubernetes cluster version: v%s.%s. git version: %s. git tree state: %s. commit: %s. platform: %s",
+		v.Major, v.Minor, v.GitVersion, v.GitTreeState, v.GitCommit, v.Platform)
+	glog.Infof("Communication with server successful")
+
+	resolver := scale.NewDiscoveryScaleKindResolver(discoveryClient)
+	cachedDiscoveryClient := cacheddiscovery.NewMemCacheClient(discoveryClient)
+	mapper := restmapper.NewDeferredDiscoveryRESTMapper(cachedDiscoveryClient)
+	scaleNamespacer := scale.New(restClient, mapper, dynamic.LegacyAPIPathResolverFunc, resolver)
```

Comment:
I'm not very familiar with this API - do you foresee any problem similar to https://github.com/kubernetes-sigs/cluster-proportional-vertical-autoscaler/issues/35? I do see the implementations are different, just not sure about how the underlying discovery will happen.